lib/ansible/modules/cloud/amazon/ec2_spot_pricing_history_facts.py

```diff
@@ -0,0 +1,226 @@
+    one_week_ago: >-
+      {{ "%Y-%m-%dT%H:%M:%SZ" | strftime((ansible_date_time.epoch | int ) - (86400 * 7 ) }}
+
+- ec2_spot_pricing_history_facts:
+    availability_zone: "eu-west-1a"
+    instance_types: ["t3.small"]
+    product_descriptions: ["Linux/UNIX"]
+    start_time: "{{ one_week_ago }}"
+    end_time: "{{ ansible_date_time.iso8601 }}"
+"""
+
+RETURN = """
+---
+ec2_spot_pricing_history:
+  description: the spot pricing history
+  returned: success
+  type: list
+  sample: [
+    {
+      "availability_zone": "eu-west-1a",
+      "spot_price": 0.71,
+      "instance_type": "t3.medium",
+      "product_description": "Linux/UNIX",
+      "timestamp": "2019-02-12T13:47:13+00:00"
+    }
+  ]
+"""
+
+import datetime
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    ec2_argument_spec,
+    get_aws_connection_info,
+    camel_dict_to_snake_dict,
+)
+
+
+def parse_date(module, key):
+    value = module.params[key]
+    if value == "now":
+        return datetime.datetime.now()
+    try:
+        return datetime.datetime.strptime(value, "%Y-%m-%dT%H:%M:%SZ")
+    except ValueError:
+        msg = ("{key}={value} is not a valid date. Dates must be specified in " +
+               "ISO8601 format")
+        module.fail_json(
+            msg=msg.format(key=key, value=value)
+        )
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            availability_zone=dict(type="str", default=""),
+            product_descriptions=dict(
+                type="list",
+                default=[],
+                choices=[
+                    "Linux/UNIX",
+                    "SUSE Linux",
+                    "Windows",
+                    "Linux/UNIX (Amazon VPC)",
+                    "SUSE Linux (Amazon VPC)",
+                    "Windows (Amazon VPC)",
+                ],
+            ),
+            instance_types=dict(type="list", default=[]),
+            filters=dict(type="list", default=[]),
+            start_time=dict(type="str", default="now"),
+            end_time=dict(type="str", default="now"),
+            max_results=dict(type="int", default=1000),
+        )
+    )
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, check_boto3=True)
+
+    if not module.boto3_at_least("1.9.0"):
+        module.fail_json(msg="ec2_spot_price_history_facts requires boto3 > 1.9.0")
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    if not region:
+        module.fail_json(msg="region must be specified")
+
+    ec2_client = boto3_conn(
+        module,
+        conn_type="client",
+        resource="ec2",
+        region=region,
+        endpoint=ec2_url,
+        **aws_connect_params
+    )
+
+    start_time = parse_date(module, "start_time")
+    end_time = parse_date(module, "end_time")
+
+    spot_price_history_response = ec2_client.describe_spot_price_history(
```

Comment:
This call should be in a try block catches `botocore.exceptions.BotoCoreError` and  `botocore.exceptions.ClientError` and calls `fail_json_aws()` when they're raised.